src/memgraph.cpp

```diff
@@ -349,39 +349,54 @@ DEFINE_VALIDATED_string(log_level, "WARNING", log_level_help_string.c_str(), {
 });
 
 namespace {
-void ParseLogLevel() {
+spdlog::level::level_enum ParseLogLevel() {
   const auto log_level = StringToEnum<spdlog::level::level_enum>(FLAGS_log_level, log_level_mappings);
   MG_ASSERT(log_level, "Invalid log level");
-  spdlog::set_level(*log_level);
+  return *log_level;
 }
 
 // 5 weeks * 7 days
 constexpr auto log_retention_count = 35;
 
-void ConfigureLogging() {
-  std::vector<spdlog::sink_ptr> loggers;
+struct MemgraphLogger {
+  MemgraphLogger() : log_level(ParseLogLevel()) {
+    std::vector<spdlog::sink_ptr> sinks;
 
-  if (FLAGS_also_log_to_stderr) {
-    loggers.emplace_back(std::make_shared<spdlog::sinks::stderr_color_sink_mt>());
-  }
+    if (FLAGS_also_log_to_stderr) {
```

Comment:
What about passing this option as a parameter of the constructor? This way I think the responsibilities are clearly separated, the logger no longer depends on the config flag.